cpp/test/core/apply_tests.cpp

```diff
@@ -1,6 +1,6 @@
 #include "precompiled.h"
 #include "apply_tests.h"
-#include "apply_test_reflection.h"
+#include "apply_test_apply.h"   // The _reflection.h should not be included.
```

Comment:
Minor: consider expanding the comment a bit indicating that we don't want to include _reflection so that we _only_ see the pre-generated apply overloads, instead of potentially expanding better ones based on what's in _reflection.h